akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGroupedWeightedSpec.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.scaladsl
+
+import akka.stream.testkit.{ ScriptedTest, StreamSpec, TestPublisher, TestSubscriber }
+import akka.testkit.TimingTest
+import akka.util.unused
+
+import scala.collection.immutable
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class FlowGroupedWeightedSpec extends StreamSpec("""
+    akka.stream.materializer.initial-input-buffer-size = 2
+  """) with ScriptedTest {
+
+  "A GroupedWeighted" must {
+    "produce empty sequence (group) when source is empty" in {
+      val input = immutable.Seq.empty
+      def costFn(@unused e: Int): Long = 999999L // set to an arbitrarily big value
+      val future = Source(input).groupedWeighted(1)(costFn).runWith(Sink.headOption)
+      val result = Await.result(future, remainingOrDefault)
+      result should be(None)
+    }
+
+    "always exhaust a source into a single group if cost is 0" in {
+      val input = (1 to 15)
+      def costFn(@unused e: Int): Long = 0L
+      val minWeight = 1 // chose the least possible value for minWeight
+      val future = Source(input).groupedWeighted(minWeight)(costFn).runWith(Sink.head)
+      val result = Await.result(future, remainingOrDefault)
+      result should be(input)
+    }
+
+    "exhaust source into one group if minWeight equals the accumulated cost of the source" in {
+      val input = (1 to 16)
+      def costFn(@unused e: Int): Long = 1L
+      val minWeight = input.length
+      val future = Source(input).groupedWeighted(minWeight)(costFn).runWith(Sink.head)
```

Comment:
I think a better verification would be to use `Sink.seq` and then also verify that there is only one group in the result, `Sink.head` would throw away any subsequent groups making the test succeed also if there was such a bug in the operator (pretty sure there isn't, but for good measure). Would be good to do for all the test cases now using `Sink.head`.

Reply:
Great point. I'll update all of them.